### DISCUSSION TITLE
How to delete a Custom Resource before deploying a new one using kustomization

### DISCUSSION THREAD

**Discussion by Tigrex** [fluxcd/flux2#3682] [Q&A] [answered]
Hi Flux experts, now I have an application based on K8s operators and kustomization.
The custom resource works like a k8s job, and doesn't support update operation (patching existing CR only updates the CR itself, and has no effect on the running or completed pod). 

In order to deploy a new custom resource due to image or env change, I need to delete the old one first.
I can do it in two steps: first delete the old CR with "$patch: delete", and then patch it with updated image/env.

I have also tried to make two kustomizations, with the first one doing the delete and the second one doing deployment, with dependencies specified. They both point to the same base but have different overlays. However, it doesn't seem to work because the two kustomization cannot point to the exact same resource.

Is there a better way to do it?

**Comment by Tigrex** [comment]
I found another potential solution. In order to support k8s Job update, which is immutable, Flux kustomization has a configuration Kustomization.spec.force that will delete the old Job before creating a new one.
Does this flag work for all immutable resources (including CRDs) or only for K8s Jobs?

Unfortunately, my CRD is defined as mutable and won't get benefit from this configuration.

**Comment by makkes** [comment, marked as answer]
Set [`.spec.prune: true`](https://fluxcd.io/flux/components/kustomize/kustomization/#garbage-collection) and use a different `.metadata.name` for the new CR. This will delete the old one and create a new one.

**Reply by Tigrex** [reply]
Thank you! I tried patchesJson6902, and it can be used to patch images, env together with .metadata.name.
